docs/specs/audit-log.md

```diff
@@ -0,0 +1,152 @@
+- [Audit Log](#audit-log)
+  * [Storage Considerations](#storage-considerations)
+    + [TimescaleDB](#timescaledb)
+    + [PostgreSQL 10 native partitioning](#postgresql-10-native-partitioning)
+    + [pg\_partman](#pg--partman)
+  * [Event List](#event-list)
+    + [user.authenticated](#userauthenticated)
+    + [user.signed_out](#usersigned-out)
+  * [Database table schema](#database-table-schema)
+  * [Admin API](#admin-api)
+
+# Audit Log
+
+Audit log records important user activities.
+
+## Storage Considerations
+
+To avoid further complicating the deployment requirements,
+audit log is stored in a separate PostgreSQL database.
+The PostgreSQL database can be a separate instance,
+or it can be within the same instance, but a different database.
+There exists a PostgreSQL extension called pg\_partman
+which greatly simplify the management of partitioned tables.
+
+Alternatively, audit log could be stored in Elasticsearch.
+However, as of Elasticsearch 7, it does not support partitioning natively.
+We need to create the index partition manually,
+write specialized query with partition in mind,
+and also drop indices that is not in the retention period.
+
+### TimescaleDB
+
+TimescaleDB is a time series database based on PostgreSQL.
+It is available as managed service of self-hosted.
+Self-hosting it seems too complicated.
+So it is not preferred.
+
+### PostgreSQL 10 native partitioning
+
+Native partitioning is supported from PostgreSQL 10 onwards.
+Since audit log is time series data,
+partition can help reduce the size of the table and its indices.
+Smaller tables and indices result in shorter insert time and query time.
+
+### pg\_partman
+
+[pg\_partman](https://github.com/pgpartman/pg_partman) is an extension to manage partition.
+
+Major cloud providers such as Google, AWS and Azure provide this extension for their managed SQL database service.
+
+- https://cloud.google.com/sql/docs/postgres/extensions#pg_partman
+- https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/PostgreSQL_Partitions.html#PostgreSQL_Partitions.pg_partman
+- https://docs.microsoft.com/en-us/azure/postgresql/concepts-extensions
+
+We still have to call a stored procedure provided by pg\_partman to perform maintenance periodically.
+That procedure creates new child tables and drop expired child tables.
+For deployment on Kubernetes, we will use CronJob to run the procedure.
+
+## Event List
+
+The event list mainly borrows from [the webhook non-block event list](./webhook.md#non-blocking-events).
+
+
+### user.authenticated
```

Comment:
Should we consider adding `user.created`?  The definition will be the same as webhook
- `user.created` for user signup / admin created user
- `user.authenticated` for user login (user signup not included)